src/language/parser.ts

```diff
@@ -82,20 +82,22 @@ export interface ParseOptions {
   noLocation?: boolean;
 
   /**
-   * @deprecated will be removed in the v17.0.0
-   *
    * If enabled, the parser will understand and parse variable definitions
    * contained in a fragment definition. They'll be represented in the
    * `variableDefinitions` field of the FragmentDefinitionNode.
    *
+   * Additionally, the parser will understand arguments passed to fragment
+   * spreads. They'll be represented in the `arguments` field of the
+   * FragmentSpreadNode.
+   *
    * The syntax is identical to normal, query-defined variables. For example:
    *
    *   fragment A($var: Boolean = false) on T  {
    *     ...
    *   }
    *
    */
-  allowLegacyFragmentVariables?: boolean;
+  allowFragmentArguments?: boolean;
```

Comment:
Since we are serious about this proposal this time, I would just remove this flag and make it parsable by default.

Reply:
Done: I added a custom validation rule to indicate that fragment arguments are not part of the spec.